server/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -333,6 +337,19 @@ private GetResult innerGetFetch(
             }
         }
 
+        // NOTE: when _ignored is requested via `stored_fields` we need to load it from doc values instead of loading it from stored fields.
+        // The _ignored field used to be stored, but as a result of supporting aggregations on it, it moved from using a stored field to
+        // using doc values.
+        if (storedFields != null && Arrays.asList(storedFields).contains(IgnoredFieldMapper.NAME)) {
+            final DocumentField ignoredDocumentField = loadIgnoredMetadataField(docIdAndVersion);
+            if (ignoredDocumentField != null) {
+                if (metadataFields == null) {
+                    metadataFields = new HashMap<>();
+                }
+                metadataFields.put(IgnoredFieldMapper.NAME, ignoredDocumentField);
+            }
+        }
```

Comment:
Should we also make sure that we don't try to load `_ignored` from stored fields as well? I guess it's either loaded from doc values or from stored fields, but not from both?

Reply:
++